contracts/service-registry/src/state.rs

```diff
@@ -36,16 +36,24 @@ pub struct Worker {
     pub service_name: String,
 }
 
-impl TryInto<Participant> for Worker {
+impl TryFrom<Worker> for Participant {
     type Error = ContractError;
 
-    fn try_into(self) -> Result<Participant, ContractError> {
-        match self.bonding_state {
-            BondingState::Bonded { amount } => Ok(Participant {
-                address: self.address,
+    fn try_from(worker: Worker) -> Result<Participant, ContractError> {
+        match worker.bonding_state {
+            BondingState::Bonded { amount } => Ok(Self {
+                address: worker.address,
                 weight: amount.try_into()?,
```

Comment:
To me, this PR should have been as simple as here making it weight to be one, with a comment explaining the reason.

Reply:
this was my original idea, but CJ wanted to have a separate struct for this. I guess it could be useful for some other devs that are developing their own provers and wanna start with constant weights first as well. I don't have a strong preference one way or the other

Reply:
Maybe, but I don't like having useless but potentially-useful code around. It's confusing. People would question why it's there every time they see it.